scipy/optimize/tests/test_linprog.py

```diff
@@ -1638,6 +1638,41 @@ def test_crossover(self):
         # there should be nonzero crossover iterations for IPM (only)
         assert_equal(res.crossover_nit == 0, self.method != "highs-ipm")
 
+    def test_sensitivies(self):
+        '''
+        Ensure lagrange multipliers are correct by comparing the derivative
+        w.r.t. b_ub/b_eq/ub/lb match the reported duals.
+        '''
+        c, A_ub, b_ub, A_eq, b_eq, bounds = very_random_gen(seed=0)
+        res = linprog(c, A_ub=A_ub, b_ub=b_ub, A_eq=A_eq, b_eq=b_eq,
+                      bounds=bounds, method=self.method, options=self.options)
+
+        _kwargs = {'b_ub': b_ub,
+                   'b_eq': b_eq,
+                   'lb': bounds[:, 0],
+                   'ub': bounds[:, 1]}
+        def _central_diff(prop: str, h: float = 1e-5) -> np.ndarray:
+            res = np.empty_like(_kwargs[prop])
+            for ii in range(len(_kwargs[prop])):
+                hi = _kwargs[prop].copy()
+                lo = _kwargs[prop].copy()
+                hi[ii] += h
+                lo[ii] -= h
+                hi_kwargs = {k: (v if k != prop else hi) for k, v in _kwargs.items()}
```

Comment:
@mdhaber You may or may not like what I've done here.  There's probably a better way to do this -- this is just the first thing I thought of

Reply:
I think I had something like this in mind, but it would be a lot simpler if you could do just forward step. Does the code I provided not run without error for you? That is just forward step.

Reply:
I went straight for replacing the random program generator with very_random_gen(), but with everything else the same the tolerances were too large to pass and I couldn't find a step size that worked for everything. I'll check again tomorrow, I agree forward step would make this easier to read

Reply:
I see. I think my original code didn't pass. I wasn't using `np.testing.assert_allclose`. Well, let me think about it. It might be best just to change the tolerances.

Reply:
Might have been an implementation issue -- I just did a quick forward step and it's working so I removed the central difference in favor of that

Reply:
?! I tested it myself today, and it didn't look to me like it would pass with simple forward step! How are we flip-flopping? 

Did you see my #12 though?

Reply:
I did not see that PR -- I apologize!  I'm surprised I didn't get an email about it, I'll have to see if there's a setting I need to toggle...

I'm happy to use whichever implementation you think is more maintainable -- I'm probably not going to mess around with it anymore other than integration if needed

Reply:
So I think what's going on here is that I'm just getting lucky. I'm using naive implements of numerical differentiation without regard for scaling of the variables whereas you were trying to take that into account. For this problem, it appears the naive method wins out, but I like the approach better in #12 , so I think going to pull that in

Reply:
I also found that absolute steps were better, but still wasn't getting within default tolerance for me. Whatever!